app/src/main/java/com/example/studentarena/fragments/FeedFragment.java

```diff
@@ -69,16 +71,27 @@ public void onClick(View v) {
         // set the adapter on the recycler view
         rvPosts.setAdapter(adapter);
         queryPosts();
+        swipeContainer = (SwipeRefreshLayout) view.findViewById(R.id.swipeContainer);
+        // Setup refresh listener which triggers new data loading
+        swipeContainer.setOnRefreshListener(new SwipeRefreshLayout.OnRefreshListener() {
+            @Override
+            public void onRefresh() {
+                queryPosts();
+            }
+        });
+        // Configure the refreshing colors
+        swipeContainer.setColorSchemeResources(android.R.color.holo_blue_bright,
+                android.R.color.holo_green_light,
+                android.R.color.holo_orange_light,
+                android.R.color.holo_red_light);
     }
 
     private void queryPosts() {
+        allPosts.clear();
```

Comment:
just calling out what you mentioned today, this doesn't quite work with #19. you said you wanted to move the `allPosts.clear()` to a better place for that PR

that may turn out to be surprisingly difficult. we start out with this:
```
Refresh_branch -> o   o <- Endless_Scroll_Branch
                  |   |
                  o   o
                   \ /
                    o <- main
                    |
                    o
```
after merging this PR we get:
```
                      o <- Endless_Scroll_Branch
                      |
            main -> o o
                    |/
                    o
                    |
                    o
```

#19 branches off before the commit that added `allPosts.clear()`. if you check that branch out, that line won't be there so you can't create a git commit that removes it.

i can think of three ways to handle it:
- hope that because #19 changes some lines around where you add `allPosts.clear()` that you get a merge conflict. that would sort of coincidentally create an opportunity to erase `allPosts.clear()` before merging #19 into main
- use the `git merge` command to create a merge commit from _main_ into _Endless_Scroll_Branch_. this is backwards from what our PRs normally do. but it would tell `Endless_Scroll_Branch` about that `allPosts.clear()` line, and then you can create another commit that deletes it.
- use the `git rebase` command to basically..., erm, replay the commits from `Endless_Scroll_Branch` as if they branch off from the new `main` rather than where they branch off from now. looking at the diagram above, instead of branching off from that commit in the middle of the `main` branch, that stack of commits would be moved to branch off the top. then your branch knows about that `allPosts.clear()` line and you can create a new commit to remove it. then you use `git push --force` to tell GitHub "yes i am basically rewriting history, i know what i'm doing".

the first or second option are gonna be easiest for us, i just explained rebasing because there are a lot of teams that prefer it and you'll probably see that vocabulary again.